examples/tree.rs

```diff
@@ -26,19 +26,27 @@ fn tree(gl: &mut QuadGl, time: f64, deep: u32, angle: f32, tall: f32) {
     gl.pop_model_matrix();
 }
 
+
+
 #[macroquad::main("Tree")]
 async fn main() {
-    let camera = Camera2D {
-        zoom: vec2(1., 1.),
-        target: vec2(0.0, 0.5),
-        ..Default::default()
-    };
+    let mut zoom: f32 = 0.7;
+    let mut zoominterp = zoom;
 
-    set_camera(camera);
     loop {
-        clear_background(GREEN);
-
-        draw_circle(0., 0., 0.03, RED);
+        let camera = Camera2D {
```

Comment:
I would rather keep examples as minimal as possible, but showcasing certain features. 
Tree example is dedicated to manipulating model matrices. All the visual goodies may be done on the user side I believe :)

Reply:
Also it is usually way easier to review PR with one certain feature in it. So it would be nice if you will split this PR into two: one just for math::clamp and another one for the example.

Reply:
Nice looking exaples make good impression to Mcqd potential users. 
Hm, is it possible to fix screen ratio by somehow updating things under the hood automatically when window is being resized?
Uneven scale factors are not seeming as commonly used feature, esp for beginners, usually evertone want just a properly displayed (not stretched) image, scaled by scalar zoom factor.